pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReader.java

```diff
@@ -34,10 +34,16 @@
 public interface RecordReader extends Closeable {
 
   /**
-   * Initializes the record reader when needed
+   * initializing recordreader with inputpath, schema and recordreader config. <br/>
+   * The implementation can chose to ignore one or more of these parameters and handle null gracefully <br/>
+   *
+   * @param inputPath absolute path to the file/directory
+   * @param schema Pinot Schema associated with the table
+   * @param recordReaderConfig config for the reader specific to the format. e.g. delimiter for csv format etc
+   * @throws Exception if the arguments are invalid
    */
-  void init(SegmentGeneratorConfig segmentGeneratorConfig)
-      throws IOException;
+  void init(@Nullable String inputPath, @Nullable Schema schema, @Nullable RecordReaderConfig recordReaderConfig)
```

Comment:
Should we make inputPath and schema mandatory?

Reply:
I could not conclude that looking at the existing implementations.

Reply:
When you call `init()`, you will always provide the input path and schema right?

Reply:
only for Parquet and ORC readers. There are some record readers which dont need schema as of today.

Reply:
I understand most record readers don't need schema as of now, but when you call the `init()`, you are not supposed to pass `null` as the schema